src/main/groovy/com/palantir/gradle/docker/PalantirDockerPlugin.groovy

```diff
@@ -72,6 +91,33 @@ class PalantirDockerPlugin implements Plugin<Project> {
                 workingDir dockerDir
                 commandLine 'docker', 'push', ext.name
             }
+
+            dockerfileZip.with {
+                from(ext.resolvedDockerfile)
+            }
+
+            // Configure docker-compose templating
+            if (ext.resolvedDockerComposeTemplate.exists()) {
+                def dockerDependencies = project.configurations.docker.resolvedConfiguration.resolvedArtifacts
+                def templateTokens = dockerDependencies.collectEntries {
+                    def version = it.moduleVersion.id
+                    [("{{${version.group}:${version.name}}}"): version.version]
+                }
+
+                generateDockerCompose.with {
+                    from(ext.resolvedDockerComposeTemplate)
+                    into(ext.resolvedDockerComposeFile.parentFile)
```

Comment:
It differs from the 'Dockerfile' semantics, but that's okay I guess. The rest of our gradle work tends to produce output results in buildDir, it's a bit strange to break that convention but I understand why you're breaking it.